src/Aspire.Hosting.Redis/RedisResource.cs

```diff
@@ -18,9 +28,20 @@ public class RedisResource(string name) : ContainerResource(name), IResourceWith
     /// </summary>
     public EndpointReference PrimaryEndpoint => _primaryEndpoint ??= new(this, PrimaryEndpointName);
 
+    /// <summary>
+    /// Gets the parameter that contains the Redis server password.
+    /// </summary>
+    public ParameterResource? PasswordParameter { get; }
+
+    /// <summary>
+    /// Gets the parameter that contains the Redis server username.
+    /// </summary>
     private ReferenceExpression ConnectionString =>
```

Comment:
Prefer this pattern https://github.com/dotnet/aspire/blob/9e2a2dfbeda1b85501b3f80ad58f50f4609a1a09/src/Aspire.Hosting.MongoDB/MongoDBServerResource.cs#L56